roles/openshift_manageiq/tasks/main.yaml

```diff
@@ -3,76 +3,56 @@
 
-- name: Copy Configuration to temporary conf
-  command: >
-    cp {{ openshift.common.config_base }}/master/admin.kubeconfig {{manage_iq_tmp_conf}}
-  changed_when: false
-
 - name: Add Management Infrastructure project
-  command: >
-    {{ openshift.common.client_binary }} adm new-project
-    management-infra
-    --description="Management Infrastructure"
-    --config={{manage_iq_tmp_conf}}
-  register: osmiq_create_mi_project
-  failed_when: "'already exists' not in osmiq_create_mi_project.stderr and osmiq_create_mi_project.rc != 0"
-  changed_when: osmiq_create_mi_project.rc == 0
+  oc_project:
+    name: management-infra
+    description: Management Infrastructure
 
 - name: Create Admin and Image Inspector Service Account
   oc_serviceaccount:
-    kubeconfig: "{{ openshift_master_config_dir }}/admin.kubeconfig"
     name: "{{ item }}"
     namespace: management-infra
     state: present
   with_items:
   - management-admin
   - inspector-admin
 
-- name: Create Cluster Role
-  shell: >
-    echo {{ manageiq_cluster_role | to_json | quote }} |
-    {{ openshift.common.client_binary }} create
-    --config={{manage_iq_tmp_conf}}
-    -f -
-  register: osmiq_create_cluster_role
-  failed_when: "'already exists' not in osmiq_create_cluster_role.stderr and osmiq_create_cluster_role.rc != 0"
-  changed_when: osmiq_create_cluster_role.rc == 0
+- name: Create manageiq cluster role
+  oc_clusterrole:
+    name: management-infra-admin
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - pods/proxy
+      verbs:
+      - "*"
 
 - name: Create Hawkular Metrics Admin Cluster Role
-  shell: >
-    echo {{ manageiq_metrics_admin_clusterrole | to_json | quote }} |
-    {{ openshift.common.client_binary }}
-    --config={{manage_iq_tmp_conf}}
-    create -f -
-  register: oshawkular_create_cluster_role
-  failed_when: "'already exists' not in oshawkular_create_cluster_role.stderr and oshawkular_create_cluster_role.rc != 0"
-  changed_when: oshawkular_create_cluster_role.rc == 0
-  # AUDIT:changed_when_note: Checking the return code is insufficient
-  # here. We really need to verify the if the role even exists before
-  # we run this task.
+  oc_clusterrole:
+    name: hawkular-metrics-admin
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - hawkular-alerts
+      - hawkular-metrics
+      verbs:
+      - "*"
 
 - name: Configure role/user permissions
-  command: >
-    {{ openshift.common.client_binary }} adm {{item}}
-    --config={{manage_iq_tmp_conf}}
-  with_items: "{{manage_iq_tasks}}"
-  register: osmiq_perm_task
-  failed_when: "'already exists' not in osmiq_perm_task.stderr and osmiq_perm_task.rc != 0"
-  changed_when: osmiq_perm_task.rc == 0
-  # AUDIT:changed_when_note: Checking the return code is insufficient
-  # here. We really need to compare the current role/user permissions
-  # with their expected state. I think we may have a module for this?
-
+  oc_adm_policy_user:
+    namespace: management-infra
```

Comment:
For some reason, this task reported `changed` on the second run (for idempotency), but it was only on the first two items:
```
2017-04-12 16:41:19,707 p=28455 u=rteague |  TASK [openshift_manageiq : Configure role/user permissions] ********************
2017-04-12 16:41:19,707 p=28455 u=rteague |  task path: /home/rteague/dev/clusters/aws-c1/openshift-ansible/roles/openshift_manageiq/tasks/main.yaml:43
2017-04-12 16:41:21,015 p=28455 u=rteague |  changed: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'role', u'resource_name': u'admin', u'user': u'management-admin'}) => {
    "changed": true, 
    "item": {
        "resource_kind": "role", 
        "resource_name": "admin", 
        "user": "management-admin"
    }, 
    "present": "present", 
    "results": {
        "cmd": "/bin/oc adm policy add-role-to-user admin management-admin -n management-infra", 
        "results": "", 
        "returncode": 0
    }
}
2017-04-12 16:41:22,047 p=28455 u=rteague |  changed: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'role', u'resource_name': u'management-infra-admin', u'user': u'management-admin'}) => {
    "changed": true, 
    "item": {
        "resource_kind": "role", 
        "resource_name": "management-infra-admin", 
        "user": "management-admin"
    }, 
    "present": "present", 
    "results": {
        "cmd": "/bin/oc adm policy add-role-to-user management-infra-admin management-admin -n management-infra", 
        "results": "", 
        "returncode": 0
    }
}
2017-04-12 16:41:22,905 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'cluster-role', u'resource_name': u'cluster-reader', u'user': u'system:serviceaccount:management-infra:management-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "cluster-role", 
        "resource_name": "cluster-reader", 
        "user": "system:serviceaccount:management-infra:management-admin"
    }, 
    "present": "present"
}
2017-04-12 16:41:23,730 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'scc', u'resource_name': u'privileged', u'user': u'system:serviceaccount:management-infra:management-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "scc", 
        "resource_name": "privileged", 
        "user": "system:serviceaccount:management-infra:management-admin"
    }, 
    "present": "present"
}
2017-04-12 16:41:24,573 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'cluster-role', u'resource_name': u'system:image-puller', u'user': u'system:serviceaccount:management-infra:inspector-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "cluster-role", 
        "resource_name": "system:image-puller", 
        "user": "system:serviceaccount:management-infra:inspector-admin"
    }, 
    "present": "present"
}
2017-04-12 16:41:25,397 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'scc', u'resource_name': u'privileged', u'user': u'system:serviceaccount:management-infra:inspector-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "scc", 
        "resource_name": "privileged", 
        "user": "system:serviceaccount:management-infra:inspector-admin"
    }, 
    "present": "present"
}
2017-04-12 16:41:26,211 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'cluster-role', u'resource_name': u'self-provisioner', u'user': u'system:serviceaccount:management-infra:management-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "cluster-role", 
        "resource_name": "self-provisioner", 
        "user": "system:serviceaccount:management-infra:management-admin"
    }, 
    "present": "present"
}
2017-04-12 16:41:27,039 p=28455 u=rteague |  ok: [ec2-54-91-22-173.compute-1.amazonaws.com] => (item={u'resource_kind': u'cluster-role', u'resource_name': u'hawkular-metrics-admin', u'user': u'system:serviceaccount:management-infra:management-admin'}) => {
    "changed": false, 
    "item": {
        "resource_kind": "cluster-role", 
        "resource_name": "hawkular-metrics-admin", 
        "user": "system:serviceaccount:management-infra:management-admin"
    }, 
    "present": "present"
}
```